packages/server-wallet/src/engine/engine.ts

```diff
@@ -544,6 +546,23 @@ export class SingleThreadedEngine
     return response.multipleChannelOutput();
   }
 
+  async approveObjectives(
+    objectiveIds: string[]
+  ): Promise<{objectives: WalletObjective[]; messages: Message[]}> {
+    const objectives = await this.store.getObjectivesByIds(objectiveIds);
+
+    const response = EngineResponse.initialize();
+    for (const objective of objectives) {
+      const {targetChannelId: channelId} = objective.data;
+      await this.registerChannelWithChainService(channelId);
+
+      await this.store.approveObjective(objective.objectiveId);
+
+      await this.takeActions([channelId], response);
```

Comment:
Should `takeActions` get invoked once with a list of all channel ids after all of the objectives have been approved?

Reply:
Updated in ed20473